stream-projects/chat/components/chat/chat.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  classKebab,
+  // urql
+  For,
+  // ExtensionInfo,
+  getClient as _getClient,
+  ObservableArray,
+  // extension
+  ObservableObject,
+  // react
+  React,
+  Switch,
+  useStyleSheet,
+} from "../../deps.ts";
+
+import { Badge, NormalizedChatMessage } from "../../shared/mod.ts";
+import ThemeComponent from "../theme-component/theme-component.tsx";
+import styleSheet from "./chat.scss.js";
+import { default as BadgeRenderer } from "../badges/badge.tsx";
+const VERIFIED_CHECK_IMG_URL =
+  "https://cdn.bio/assets/images/features/browser_extension/yt_check_white.svg";
+
+export default function Chat(
+  { messages$ }: { messages$: ObservableArray<NormalizedChatMessage[]> },
+) {
+  useStyleSheet(styleSheet);
+
+  // TODO: use this to render a scroll to bottom button
+  const handleScroll = (e: React.UIEvent<HTMLDivElement>) => {
+    const scrollTop = (e.target as HTMLDivElement).scrollTop;
+    const isPinnedToBottom = scrollTop < 0;
+    if (isPinnedToBottom) {
+      console.log("not bottom");
+    } else {
+      console.log("pinned to bottom");
+    }
+  };
+
+  return (
+    <div className="c-chat">
+      <ThemeComponent />
+      <div className="messages">
+        <div className="inner" onScroll={handleScroll}>
+          <For<NormalizedChatMessage, {}>
+            each={messages$}
+            item={ChatMessage}
+            optimized
+          />
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export function ChatMessage(
+  { item }: { item: ObservableObject<NormalizedChatMessage> },
+) {
+  return (
+    <Switch value={item.type}>
+      {{
+        "text": () => (
```

Comment:
Rendering logic that corresponds to a messages `type` will live in this switch (e.g superchats, system messages, polls, etc.)